keras_cv/models/object_detection/yolox/layers/csp_darknet_utils_legacy.py

```diff
@@ -0,0 +1,376 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""CSPDarkNet model utils for KerasCV.
```

Comment:
If we're going to keep these "legacy" utils around, let's add a TODO to get rid of them ASAP.

Is it a big lift to update YOLOx to use the new `apply` blocks instead?

Reply:
Sure, I will add a TODO. this one need to stay a little bit longer till the new yolox is in.
Currently, it is not easy to switch to the `apply_` functions given the current implementation of yolox.
However, I believe it would be easy to do after the yolox has been migrated.